### PR TITLE
Adjust the KillService to handle instances instead of tasks

### LINKED ISSUE #4394: Cleanup/rename TaskKillService

### PR DESCRIPTION
**Pull request by meichstedt**
Resolves #4394

Note the renamings:
TaskKillProgressActor -> InstanceKillProgressActor
TaskKillServiceActor -> KillServiceActor
TaskKillServiceDelegate -> KillServiceDelegate

unfortunately GH doesn't recognize them although edit/rename actions are in separate commits.

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala [hunk 1]
```diff
@@ -606,6 +606,15 @@ trait EventFormats {
       "eventType" -> change.eventType
     )
   }
+  implicit lazy val UnknownInstanceTerminatedEventWrites: Writes[UnknownInstanceTerminated] = Writes { change =>
+    Json.obj(
+      "instanceId" -> change.id,
+      "runSpecId" -> change.runSpecId,
+      "instanceStatus" -> change.status.toString,
```
Comment (jasongilanfarr) [inline src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala]: runSpecVersion?

Comment (meichstedt) [inline src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala]: We don't have that at hand when the task is unknown, we have only the mesos TaskStatus and that doesn't provide the runSpec version safely (there might be a label iirc)

**Comment by aquamatthias** [comment]
retest this please

File: src/main/scala/mesosphere/marathon/core/task/termination/KillService.scala [hunk 2]
```diff
@@ -19,7 +19,7 @@ trait TaskKillService {
     * @param reason the reason why the task shall be killed.
     * @return a future that is completed when all tasks are killed.
     */
-  def killTasks(tasks: Iterable[Instance], reason: TaskKillReason): Future[Done]
+  def killTasks(tasks: Iterable[Instance], reason: KillReason): Future[Done]
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/core/task/termination/KillService.scala]: rename to killInstances?

File: src/main/scala/mesosphere/marathon/core/task/termination/KillService.scala [hunk 3]
```diff
@@ -29,7 +29,7 @@ trait TaskKillService {
     * @param reason the reason why the task shall be killed.
     * @return a future that is completed when all tasks are killed.
     */
-  def killTask(task: Instance, reason: TaskKillReason): Future[Done]
+  def killTask(task: Instance, reason: KillReason): Future[Done]
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/core/task/termination/KillService.scala]: rename to killInstance?

File: src/main/scala/mesosphere/marathon/core/task/termination/impl/Terminal.scala [hunk 4]
```diff
@@ -1,18 +1,23 @@
 package mesosphere.marathon.core.task.termination.impl
 
-import mesosphere.marathon.core.event.MesosStatusUpdateEvent
+import mesosphere.marathon.core.event.InstanceChanged
+import mesosphere.marathon.core.instance.InstanceStatus
 
+// TODO(PODS): There are various similar Terminal extractors, Sets and functions – the NEED to be aligned
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/core/task/termination/impl/Terminal.scala]: There is already a Terminal trait and extractor, that should reflect exactly those cases - or?

**Comment by aquamatthias** [comment]
LGTM!

**Pull request merged**